### ISSUE TITLE
Escargot documentation

### ISSUE THREAD

**Issue by shetzl** [gapt/gapt#592]
There is currently virtually no documentation of the escargot prover, in particular there is no scaladoc documentation. Gabriel, please add suitable scaladoc comments explaining at least the overall structure of the prover. If I remember correctly, the original design was based on a Handbook-article, maybe it makes sense to cite that and explain the prover w.r.t. this article.

**Comment by gebner** [comment]
The first version was based somewhat closely on some chapters in the Handbook.  But at that time I didn't really knew what I was doing, and just kept adding stuff whenever it failed to prove something.  Since then, I've refactored Escargot quite a bit, it includes a few techniques that are much newer than the Handbook.  Right now I'm quite busy, I'll add scaladocs when I have time.

**Comment by shetzl** [comment]
Sure, it's not urgent.

**Comment by gebner** [comment]
It was based on a mix of various Handbook chapters in the beginning, as I didn't really knew what I was doing and just cobbled together parts when it failed to prove a theorem.  Since then, I've refactored Escargot quite a bit and it includes techniques that are much newer than the Handbook.
I'm quite busy right now, but I'll write scaladocs when I've got more time.  I hope this issue isn't too pressing.
